### PR TITLE
added conditional get (ifModifiedSince) to scraper

### PR DESCRIPTION
**Pull request by ymatsuda**

### CODE REVIEW

File: shoebox/app/com/keepit/model/ScrapeInfo.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ import org.specs2.internal.scalaz.FirstOption
 case class ScrapeInfo(
   id: Option[Id[ScrapeInfo]] = None,
   uriId: Id[NormalizedURI], // = NormalizedURI id
-  lastScrape: DateTime = currentDateTime,
+  lastScrape: DateTime = START_OF_TIME,
```
Comment (eishay) [inline shoebox/app/com/keepit/model/ScrapeInfo.scala:23]: :-)

File: shoebox/app/com/keepit/scraper/HttpFetcher.scala [hunk 2]
```diff
@@ -43,14 +45,21 @@ class HttpFetcher extends Logging {
...
-  def fetch(url: String)(f: HttpInputStream => Unit): HttpFetchStatus = {
-    val httpget = new HttpGet(url)
-    log.info("executing request " + httpget.getURI())
+  def fetch(url: String, ifModifiedSince: Option[DateTime] = None)(f: HttpInputStream => Unit): HttpFetchStatus = {
+    val httpGet = new HttpGet(url)
 
-    val httpContext = new BasicHttpContext();
-    val response = httpClient.execute(httpget, httpContext)
+    ifModifiedSince.foreach{ ifModifiedSince =>
+      httpGet.addHeader("If-Modified-Since", ifModifiedSince.format)
```
Comment (eishay) [inline shoebox/app/com/keepit/scraper/HttpFetcher.scala:52]: Cool, how about using the constant http://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/HttpHeaders.html#IF_MODIFIED_SINCE i think we have it in the classpath

Comment (ymatsuda) [inline shoebox/app/com/keepit/scraper/HttpFetcher.scala:52]: Thanks! I will do so.

File: shoebox/app/com/keepit/scraper/HttpFetcher.scala [hunk 3]
```diff
@@ -93,8 +103,17 @@ class HttpFetcher extends Logging {
...
       }
     } else {
-      httpget.abort();
-      HttpFetchStatus(-1, Some("no entity found"), httpContext)
+      httpGet.abort()
+      statusCode match {
+        case HttpStatus.SC_OK =>
+          log.info("request failed: [%s][%s]".format(response.getStatusLine().toString(), url))
+          HttpFetchStatus(-1, Some("no entity found"), httpContext)
+        case HttpStatus.SC_NOT_MODIFIED =>
```
Comment (eishay) [inline shoebox/app/com/keepit/scraper/HttpFetcher.scala:111]: great!

**Pull request merged**